src/entity.ts

```diff
@@ -326,11 +326,67 @@ export namespace entity {
     return value instanceof entity.Key;
   }
 
+  /**
+   * Convert a protobuf `integerValue`.
+   *
+   * @private
+   * @param {object} valueProto The protobuf `integerValue` to convert.
+   * @param {object} typeCastConfig Config for custom `integerValue` cast.
+   * @property {function} {typeCastConfig.typeCastFunction} A custom user
+   *     provided function to convert `integerValue`.
+   * @property {sting|string[]} {typeCastConig.names} `Entity` property
+   *     names to be converted using `typeCastFunction`.
+   */
+  function decodeIntegerValue(
+    valueProto: ValueProto,
+    typeCastConfig?: TypeCastConfig
+  ) {
+    let customCast = false;
+    if (typeCastConfig) {
+      if (
+        !typeCastConfig.typeCastFunction ||
```

Comment:
Just a nit, but you could just do the `typeof` check here.

Reply:
True, thanks
Done